Source/WebCore/platform/text/TextFlags.cpp

```diff
@@ -278,13 +264,60 @@ FeaturesMap computeFeatureSettingsFromVariants(const FontVariantSettings& varian
         break;
     }
 
+    // We don't do anything if "normal".
     if (!variantSettings.alternates.isNormal()) {
         auto values = variantSettings.alternates.values();
         if (values.historicalForms)
             features.set(fontFeatureTag("hist"), 1);
         
-        // TODO: handle other tags.
-        // https://bugs.webkit.org/show_bug.cgi?id=246121
+        if (fontFeatureValues) {
+            auto lookupTags = [](const auto& name, const auto& tags) -> Vector<int> {
+                if (!name)
+                    return { };
+
+                auto found = tags.find(*name);
+                if (found == tags.end())
+                    return { };
+
+                return found->value;
+            };
+
+            auto addFeatureTagWithValue = [&features, &lookupTags] (const auto& name, const auto& tags, auto codename) {
+                for (const auto& value : lookupTags(name, tags)) {
+                    // The spec says that we ignore value greater than 99.
+                    if (value > 99)
```

Comment:
It’s unclear what prevents this from being `< 0`.

Reply:
Indeed, it should be `unsigned` from the beginning. I've fixed it.